### PR TITLE
style: only check for unstaged changes once

### PR DESCRIPTION
**Pull request by ppannuto**
A bit copy/pasty, but that's fine for a one-off like this IMO

Resolves Problem 1 from #432.

### CODE REVIEW

**Comment by ppannuto** [comment]
@brghena sorry for the slower turn on round two.., had to think a bit more, but I think this should really do what we want now.

**Comment by brghena** [comment]
What's the magic that makes this work? Is that Make knows to only run the dependency rule once?

**Comment by ppannuto** [comment]
Yep, all of the formatting stuff depends on the same phony order-only dependency, so make will only run it once

File: userland/AppMakefile.mk [hunk 1]
```diff
@@ -206,16 +206,16 @@ $(BUILDDIR)/format:
 .PHONY: fmt format
 fmt format:: $(FORMATTED_FILES)
 
-$(BUILDDIR)/format/%.uncrustify: %.c
+$(BUILDDIR)/format/%.uncrustify: %.c | _format_check_unstaged
```
Comment (brghena) [inline userland/AppMakefile.mk]: Shouldn't there be a rule for header files too?

**Comment by ppannuto** [comment]
Eventually, yes. The current version of uncrustify has that bug we talked
about a little while ago in lab where it doesn't always track #ifdef's
correctly, so it barfs on some of our headers :(

**Pull request merged**